### ISSUE TITLE
[quarkus-main] BigQuery native compilation fails due to unresolved field io.netty.buffer.PoolArena.chunkSize

### ISSUE THREAD

**Issue by jamesnetherton** [apache/camel-quarkus#5641]
### Bug description

Quarkus has upgraded Netty to `4.1.103.Final`. The version of Apache Arrow used by `google-cloud-biqquery`is not compatible with this. See https://github.com/apache/arrow/issues/39265.

**Comment by jamesnetherton** [comment]
The workaround for now was to copy the updated Arrow Netty classes to the BigQuery extension. However, give the simple nature of the original fix:

https://github.com/apache/arrow/commit/a4a3d3f4825eb025657121e70c9d86e8d6ecff35

I wonder if we could have a `BuildStep` to do some bytecode hackery and avoid having to carry the additional sources. It could even be activated conditionally based on the version of arrow that we detect.

**Comment by zhfeng** [comment]
@jamesnetherton any hint to create a tranformer of bytecode? I tried such things before but it seems hard to work.

**Comment by jamesnetherton** [comment]
> @jamesnetherton any hint to create a tranformer of bytecode? I tried such things before but it seems hard to work.

Yeah, it's tricky. I was actually just thinking about this and whether it's worth trying to do it.

`java-bigquery` has upgraded to Arrow 15.x now:

https://github.com/googleapis/java-bigquery/commit/4d4cbae00028f84c071f51e88d7ee976efdf04d5

So maybe we'll get a release with that change incorporated soon and we could remove the hacks.
